### ISSUE TITLE
TimeZone::datetime_from_str incorrectly fails to parse UNIX timestamps as Impossible

### ISSUE THREAD

**Issue by alexchandel** [chronotope/chrono#194]
The following code fails to parse with `ParseError(Impossible)` when it shouldn't:

```rust
Local.datetime_from_str("1510427171.572982", "%s%.f");
```

There's clearly enough information to parse `1510427171.572982` to a unique datetime in the `Local` timezone, yet `chrono` fails. If `Local.timestamp(0, 0)` produces a valid `DateTime<Local>`, then there's no reason this example shouldn't as well.

**Comment by alexchandel** [comment]
The documentation additionally claims "See also `Offset::datetime_from_str` which gives a local DateTime on specific time zone", but that doesn't seem to exist.

**Comment by alexchandel** [comment]
In fact, it appears to be impossible to parse a `&str` reading `1510427171.572982` into any `DateTime` at all in `chrono` without reallocating & editing the string.

**Comment by alexchandel** [comment]
Even `NaiveDateTime`'s `parse_from_str`, which ac48f514bc8e82bb146980352f0fb72b083e923f claims `does not require a timezone in the to-be-parsed str`, still fails with `ParseError(Impossible)`.

```rust
let ts = NaiveDateTime::parse_from_str("1510427171.572982", "%s%.f")?;
Local.from_utc_datetime(&ts)
```

**Comment by alexchandel** [comment]
Attempting to manually parse a timestamp, `Parsed::to_naive_date` and `Parsed::to_naive_time` both report `NotEnough`.

However, `Parsed::to_naive_datetime_with_offset` appears to incorrectly assume that the subsecond value must be zero because a timestamp is present. This is incorrect.

**Comment by quodlibetor** [comment]
This is indeed a bug in the released chrono (dupe of #175 ), and is fixed in master (5124c0c324). If you want to use the updated chrono locally (until @lifthrasiir can release a 0.4.1) I recommend using a `[patch]` section in your cargo.toml:

```toml
[patch]
chrono = { git = "https://github.com/chronotope/chrono" }
```

should do it.
